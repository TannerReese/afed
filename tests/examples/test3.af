# Tests call parsing

# Test array indexing
my_arr: [1/2, 2, 3, -1],
my_arr (1-1*my_arr 2/3)+1 = ``,
my_arr 80 = ``,
# Test repeated calls
[2, [0, [8, 1, [42, 3]]]] 1 1 2 0 = `42`,

# Test map indexing
my_map: {"a": 3/4, "b": -1.2, "xc": 4, "d": "x"},
[my_map "xc"*3, 1+my_map "a"] = ``,
my_map "z" = ``,

# Test one-variable builtin functions
sq: sqrt,
sq 3 = ``,
num.sqrt 5 = ``,
sq (-3) = ``,
# Test function application operator
tanh $ asinh $ sin $ pi / 4 = ``,
exp $ log10 $ log 3 40 = ``,

# Test two-variable builtin functions
gcd (-30) 42 = ``,
lcm (-96) 126 = ``,
num.gcd (14 / 69) (-81 / 92) = ``,

# Test partial currying
(gcd 75) 60 = ``,
least: lcm 24 = ``, # Test function printing
least 100 = ``,

