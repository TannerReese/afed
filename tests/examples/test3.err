line 6, column 13 Eval Error: Index 80 is out of bounds
line 13, column 14 Eval Error: Key z is not contained in map
line 18, column 13 Eval Error: Cannot take square root of negative
3 Eval Errors encountered
