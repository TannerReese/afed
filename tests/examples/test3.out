# Tests call parsing

# Test array indexing
my_arr: [1/2, 2, 3, -1],
my_arr (1-1*my_arr 2/3)+1 = `3 / 2`,
my_arr 80 = `Eval Error: Index 80 is out of bounds`,
# Test repeated calls
[2, [0, [8, 1, [42, 3]]]] 1 1 2 0 = `42`,

# Test map indexing
my_map: {"a": 3/4, "b": -1.2, "xc": 4, "d": "x"},
[my_map "xc"*3, 1+my_map "a"] = `[12, 7 / 4]`,
my_map "z" = `Eval Error: Key z is not contained in map`,

# Test one-variable builtin functions
sq: sqrt,
sq 3 = `1.7320508075688772`,
num.sqrt 5 = `2.23606797749979`,
sq (-3) = `Eval Error: Cannot take square root of negative`,
# Test function application operator
tanh $ asinh $ sin $ pi / 4 = `0.5773502691896256`,
exp $ log10 $ log 3 40 = `1.6922347808648706`,

# Test two-variable builtin functions
gcd (-30) 42 = `6`,
lcm (-96) 126 = `-2016`,
num.gcd (14 / 69) (-81 / 92) = `1 / 276`,

# Test partial currying
(gcd 75) 60 = `15`,
least: lcm 24 = `(num.lcm) (24)`, # Test function printing
least 100 = `600`,

